### PR TITLE
Refactor messagebody + subset of markdown

### PR DESCRIPTION
**Pull request by Simon-Laux**
(`**bold**`,`*italics*`,`~~striketrough~~`,`__underline__`)
(**bold**,*italics*,~~striketrough~~,__underline__)

Closes https://github.com/deltachat/deltachat-desktop/issues/491

### CODE REVIEW

File: src/renderer/components/conversations/Message.js [hunk 1]
```diff
@@ -206,86 +205,38 @@ class Message extends React.Component {
...
+  //       authorAddress={quote.authorAddress}
+  //       authorProfileName={quote.authorProfileName}
+  //       authorName={quote.authorName}
+  //       authorColor={quote.authorColor}
+  //       referencedMessageNotFound={quote.referencedMessageNotFound}
+  //       isFromMe={quote.isFromMe}
+  //       withContentAbove={withContentAbove}
+  //     />
+  //   )
+  // }
```
Comment (ralphtheninja) [inline src/renderer/components/conversations/Message.js]: If it's not used, throw it away :)

File: src/renderer/components/MessageBody.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class MessageBody extends React.Component {
+  render () {
+    const { text, disableJumbomoji, preview } = this.props
+    // if text is only emojis and Jumbomoji is enabled
+    const emojifiedText = text.replace(/:[\w\d_\-+]*:/g, replaceColons)
+    if (
+      emojifiedText.length < 50 &&
+            !disableJumbomoji &&
+            emojifiedText.replace(emojiRegex, '').trim() === ''
```
Comment (Jikstra) [inline src/renderer/components/MessageBody.js]: Can you align those lines?

File: src/renderer/components/conversations/Message.js (hunk 1 above)
Comment (Jikstra) [reply src/renderer/components/conversations/Message.js]: Yup throw it away :)

**Comment by Jikstra** [comment]
Very nice! And finally some markdown :)

**Comment by ralphtheninja** [comment]
@Simon-Laux Would it be hard to add support for backticks?

**Comment by Simon-Laux** [comment]
do you mean code backticks like ``` and `?
```
code block
```
`code snippet`
@ralphtheninja ?
It's supper easy with the library I added for it, it only comes down to styling, thats why I dind't included it already.

File: src/renderer/components/MessageBody.js (hunk 2 above)
Comment (Simon-Laux) [reply src/renderer/components/MessageBody.js]: sure

**Comment by ralphtheninja** [comment]
> do you mean code backticks like ``` and `?

Exactly. Maybe we can add them later.

File: src/renderer/components/MessageMarkdown.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      return { content: capture[1] }
+    },
+    react: function (node, output, state) {
+      return <a href={node.content} key={state.key}>{node.content}</a>
+    }
+  }
+  // new mailto (open chat in dc?)
+  // blockQuote ? (requires css; could be used for replies)
+  // mentions? (this component requires somekind of access to the chat memberlist)
+  // code (requires good css to look right - idea is to provide somthing without markdown formating)
```
Comment (Simon-Laux) [inline src/renderer/components/MessageMarkdown.js]: @ralphtheninja

**Comment by Simon-Laux** [comment]
I'll quickly add a things and fix a thing

**Comment by Simon-Laux** [comment]
Ok there is a bug again..

**Pull request merged**